src/Debugging/Debugger.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Debugging;
+
+use InvalidArgumentException;
+use Saloon\Contracts\DebuggingDriver;
+use Saloon\Debugging\Drivers\RayDebugger;
+use Saloon\Debugging\Drivers\ErrorLogDebugger;
+use Saloon\Debugging\Drivers\SystemLogDebugger;
+use Saloon\Exceptions\UnknownDriverException;
+
+class Debugger
+{
+    /**
+     * Globally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected static array $globalRegisteredDrivers = [];
+
+    /**
+     * Locally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected array $registeredDrivers = [];
+
+    /**
+     * Drivers that have been subscribed to
+     *
+     * @var array<string, bool>
+     */
+    protected array $useDrivers = [];
+
+    /**
+     * Denotes if we send the request to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showRequest = false;
+
+    /**
+     * Denotes if we send the response to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showResponse = false;
+
+    public function __construct()
+    {
+        $this->registerDriver(new RayDebugger);
+        $this->registerDriver(new ErrorLogDebugger);
+        $this->registerDriver(new SystemLogDebugger);
+    }
+
+    /**
+     * Register a driver globally
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return $this
+     */
+    public static function registerGlobalDriver(DebuggingDriver $driver): static
+    {
+        static::$globalRegisteredDrivers[$driver->name()] = $driver;
+
+        return new static;
+    }
+
+    /**
+     * Register a driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return $this
+     */
+    public function registerDriver(DebuggingDriver $driver): static
+    {
+        $this->registeredDrivers[$driver->name()] = $driver;
+
+        return $this;
+    }
+
+    /**
+     * Subscribe to a given driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver|string $driver A DebuggingDriver or the name one of a registered one.
+     *
+     * @return $this
+     * @throws \Saloon\Exceptions\UnknownDriverException
+     */
+    public function usingDriver(DebuggingDriver|string $driver): static
+    {
+        if ($driver instanceof DebuggingDriver) {
+            $this->registerDriver($driver);
+        }
+
+        $driverName = is_string($driver) ? $driver : $driver->name();
+
+        // We'll validate that the driver exists
+
+        $registeredDriverNames = array_keys($this->getRegisteredDrivers());
+
+        if (! in_array($driverName, $registeredDriverNames, true)) {
+            throw new UnknownDriverException(sprintf('Unable to find the "%s" driver. Registered drivers: %s', $driverName, implode(',', $registeredDriverNames)));
```

Comment:
Just a small output-suggestion.

Current: driver1,driver2,driver3
Suggested: driver1, driver2, driver3


```suggestion
            throw new UnknownDriverException(sprintf('Unable to find the "%s" driver. Registered drivers: %s', $driverName, implode(', ', $registeredDriverNames)));
```

Reply:
Great spot